src/components/oscal-utils/OSCALRestUtils.js

```diff
@@ -0,0 +1,214 @@
+      editedFieldJsonPath,
+      newValue,
+      appendToLastFieldInPath
+    );
+  } else {
+    populatePartialRestData(
+      data[editedFieldJsonPath.shift()],
+      editedFieldJsonPath,
+      newValue,
+      appendToLastFieldInPath
+    );
+  }
+}
+
+export function buildRestRequestUrl(
+  partialRestData,
+  restUrlPath,
+  oscalObjectType
+) {
+  let url;
+  if (!restUrlPath || restUrlPath === "") {
+    url = `${process.env.REACT_APP_REST_BASE_URL}/${oscalObjectType.restPath}/${partialRestData[oscalObjectType.jsonRootName].uuid}`;
+  } else if (restUrlPath.startsWith("http", 0)) {
+    url = restUrlPath;
+  } else {
+    url = `${process.env.REACT_APP_REST_BASE_URL}/${restUrlPath}`;
+  }
+  return url;
+}
+
+/**
+ * Sends a REST request of type restMethod to a backend service.
+ *
+ * @param restJsonPayload data that will be passed into the body of the REST request
+ * @param httpMethod the HTTP request type
+ * @param requestUrl the REST URL to send the request to
+ * @param onPreRestRequest function called just before making the REST request
+ * @param onSuccess function called on a successful REST request with the result of the request as an argument
+ * @param onError function called on error with the error as an argument
+ */
+export function performRestRequest(
+  restJsonPayload,
+  httpMethod,
+  requestUrl,
+  onPreRestRequest,
+  onSuccess,
+  onError
+) {
+
+  onPreRestRequest();
+
+  const requestInfo = {
+    method: httpMethod,
+    headers: {
+      "Content-Type": "application/json",
+      Accept: "application/json",
+    },
+    body: JSON.stringify(restJsonPayload),
+  };
+
+  fetch(requestUrl, requestInfo)
+    .then(
+      (response) => {
+        if (!response.ok) throw new Error(response.status);
+        else return response.json();
+      },
+      (err) => onError(err)
+    )
+    .then(
+      (result) => {
+        onSuccess(result);
+      },
+      (err) => onError(err)
+    );
+};
+
+/**
+ * Creates and performs the REST request for updating an SSP control implementation
+ * implemented requirement statement by compontent from the given data.
+ *
+ * @param partialRootRestData the partial root SSP object
+ * @param implementedRequirement the existing implemented requirement object
+ * @param statementId the id of the statement to be updated
+ * @param componentId the id of the by-component to be updated
+ * @param descriptionReference reference to the text field input containing the new description for the control implementation
+ * @param implementationSetParameters an array of the new set-parameters
+ * @param onPreRestRequest function called just before making the REST request
+ * @param onSuccess function called on a successful REST request with the result of the request as an argument
+ * @param onError function called on error with the error as an argument
+ */
+ export function updateSspControlImplementationImplementedRequirementStatementByComponent(
```

Comment:
This isn't yet called from anywhere, but shows how we could continue the pattern of isolating the model/data manipulation and REST calls from the rendering, where this would be called from `OSCALControlPartEditor`.